tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalDivisionSuite.scala

```diff
@@ -0,0 +1,232 @@
+      })
+
+      val schema = StructType(Seq(
+        StructField("c_ym", YearMonthIntervalType()),
+        StructField("c_f", FloatType),
+        StructField("c_d", DoubleType)))
+      spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+    }
+  ) {
+    df => df.selectExpr("c_ym / c_f", "c_ym / c_d")
+  }
+
+  testSparkResultsAreEqual("test year-month interval / int num, normal case",
+    spark => {
+      val numRows = 1024
+      val r = new Random(0)
+
+      def getSign: Int = if (r.nextBoolean()) 1 else -1
+
+      val data = (0 until numRows).map(i => {
+        Row(Period.ofMonths(getSign * r.nextInt(Int.MaxValue)),
+          getSign * r.nextInt(1024 * 1024) + 1, // add 1 to avoid dividing 0
+          (getSign * r.nextInt(1024 * 1024) + 1).toLong)
+      })
+
+      val schema = StructType(Seq(
+        StructField("c_ym", YearMonthIntervalType()),
+        StructField("c_i", IntegerType),
+        StructField("c_l", LongType)))
+      spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+    }
+  ) {
+    df => df.selectExpr("c_ym / c_i", "c_ym / c_l")
+  }
+
+  // both gpu and cpu will throw ArithmeticException
+  def testDivideYMOverflow(testCaseName: String, months: Int, num: Any): Unit = {
+    testBothCpuGpuExpectedException[SparkException](testCaseName + ", cv / scalar",
+      e => e.getMessage.contains("ArithmeticException"),
+      spark => {
+        val data = Seq(Row(Period.ofMonths(months)))
+        val schema = StructType(Seq(
+          StructField("c1", YearMonthIntervalType())))
+        spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+      }
+    ) {
+      df =>
+        num match {
+          case Byte => df.selectExpr(s"c1 / cast('$num' as Byte)")
+          case Short => df.selectExpr(s"c1 / cast('$num' as Short)")
+          case Int => df.selectExpr(s"c1 / $num")
+          case Long => df.selectExpr(s"c1 / ${num}L")
+          case Float.NaN => df.selectExpr("c1 / cast('NaN' as float)")
+          case Float.PositiveInfinity => df.selectExpr("c1 / cast('Infinity' as float)")
+          case Float.NegativeInfinity => df.selectExpr("c1 / cast('-Infinity' as float)")
+          case Float => df.selectExpr(s"c1 / ${num}f")
+          case Double.NaN => df.selectExpr("c1 / cast('NaN' as double)")
+          case Double.PositiveInfinity => df.selectExpr("c1 / cast('Infinity' as double)")
+          case Double.NegativeInfinity => df.selectExpr("c1 / cast('-Infinity' as double)")
+          case Double => df.selectExpr(s"c1 / ${num}d")
+          case _ => fail(s"Unsupported num type ${num.getClass}")
+        }
+    }
+
+    testBothCpuGpuExpectedException[SparkException](testCaseName + ", cv / cv",
+      e => e.getMessage.contains("ArithmeticException"),
+      spark => {
+        // Period is the external type of year-month type
+        val (data, schema) = num match {
+          case Byte => (Seq(Row(Period.ofMonths(months), num)), StructType(Seq(
```

Comment:
These cases will never match, needs to be something like this instead:
```suggestion
          case n: Byte => (Seq(Row(Period.ofMonths(months), n)), StructType(Seq(
```

Reply:
Updated.